internal/http/services/owncloud/ocdav/propfind.go

```diff
@@ -88,22 +97,49 @@ func (s *svc) handlePropfind(w http.ResponseWriter, r *http.Request, ns string)
 
 	info := res.Info
 	infos := []*provider.ResourceInfo{info}
-	if info.Type == provider.ResourceType_RESOURCE_TYPE_CONTAINER && listChildren {
-		req := &provider.ListContainerRequest{
-			Ref: ref,
-		}
-		res, err := client.ListContainer(ctx, req)
-		if err != nil {
-			log.Error().Err(err).Msg("error sending list container grpc request")
-			w.WriteHeader(http.StatusInternalServerError)
-			return
-		}
-		if res.Status.Code != rpc.Code_CODE_OK {
-			log.Err(err).Msg("error calling grpc list container")
-			w.WriteHeader(http.StatusInternalServerError)
-			return
+	if info.Type == provider.ResourceType_RESOURCE_TYPE_CONTAINER && depth != "0" {
+		// use a stack to explore sub-containers breadth-first
```

Comment:
Not an important but I just noticed, since a stack is used, it seems like sub-containers have explored in depth-first order.

Reply:
it's not, see the example result above.

there's code further down that processes the results in reverse order to make it happen that way

Reply:
I noticed that code firstly lists children of the current node. But,  since `/test/a/a1/a11/`(depth 3)  is listed before `/test/b/b1/` (depth 2) in the result, traversing across node is depth-first. So, if  `/test/a/a1/a11/a111/a1111/` (depth 5) exists, I expect it would still be listed before  `/test/b/b1/`.

I do not know, which listing order is better, maybe the current result. However, there should be a queue for breadth-first search traveling.

Reply:
right. at this stage the result order isn't that important as we have no consumer with specific expectations